assets/js/modules/adsense/util/status.test.js

```diff
@@ -0,0 +1,390 @@
+/**
+ * Status utility tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	determineAccountStatus,
+	determineSiteStatus,
+	determineAccountID,
+	determineClientID,
+	ACCOUNT_STATUS_NONE,
+	ACCOUNT_STATUS_DISAPPROVED,
+	ACCOUNT_STATUS_GRAYLISTED,
+	ACCOUNT_STATUS_MULTIPLE,
+	ACCOUNT_STATUS_NO_CLIENT,
+	ACCOUNT_STATUS_PENDING,
+	ACCOUNT_STATUS_APPROVED,
+	SITE_STATUS_NONE,
+	SITE_STATUS_ADDED,
+} from './status';
+
+const accountA = {
+	creationTime: '1382025063000',
+	id: 'pub-2833782679114991',
+	kind: 'adsense#account',
+	name: 'Test Account A',
+	premium: false,
+	timezone: 'Europe/Berlin',
+};
+
+const accountB = {
+	creationTime: '1382025063000',
+	id: 'pub-2833782711924655',
+	kind: 'adsense#account',
+	name: 'Test Account B',
+	premium: false,
+	timezone: 'Europe/Berlin',
+};
+
+const afcClientA = {
+	arcOptIn: false,
+	id: 'ca-pub-2833782679114991',
+	kind: 'adsense#adClient',
+	productCode: 'AFC',
+	supportsReporting: true,
+};
+
+const afsClientA = {
+	arcOptIn: false,
+	id: 'ca-pub-2833782679114991',
+	kind: 'adsense#adClient',
+	productCode: 'AFS',
+	supportsReporting: true,
+};
+
+const afcClientB = {
+	arcOptIn: false,
+	id: 'ca-pub-2833782711924655',
+	kind: 'adsense#adClient',
+	productCode: 'AFC',
+	supportsReporting: true,
+};
+
+const graylistedAlert = {
+	id: 'GRAYLISTED_PUBLISHER',
+	isDismissible: false,
+	kind: 'adsense#alert',
+	message: 'Your account has been graylisted. Please fix the issues in your AdSense panel.',
+	severity: 'SEVERE',
+	type: 'GRAYLISTED_PUBLISHER',
+};
+
+const otherAlert = {
+	id: 'ADS_TXT_ISSUES',
+	isDismissible: false,
+	kind: 'adsense#alert',
+	message: 'Please fix the problems with the ads.txt file to avoid negative impact on revenue.',
+	severity: 'SEVERE',
+	type: 'ADS_TXT_ISSUES',
+};
+
+const exampleURLChannelA = {
+	id: 'ca-pub-2833782679114991:example.com',
+	kind: 'adsense#urlChannel',
+	urlPattern: 'example.com',
+};
+
+const otherURLChannelA = {
+	id: 'ca-pub-2833782679114991:other-website.org',
+	kind: 'adsense#urlChannel',
+	urlPattern: 'other-website.org',
+};
+
+describe( 'determineAccountStatus', () => {
+	it( 'returns correct status based on parameters', () => {
+		const testCases = [
```

Comment:
Thanks for putting these inside the `describe` block and at least a bit closer to the actual test cases they run on (right below). It's a small thing but I find it really helps to connect the inputs/outputs to the test logic as done here. 👍